test/e2e/page-objects/pages/settings/shield/shield-detail-page.ts

```diff
@@ -0,0 +1,273 @@
+    '.transaction-shield-page__notification-banner';
+
+  private readonly membershipErrorBanner =
+    '[data-testid="membership-error-banner"]';
+
+  // Modal elements
+  private readonly cancelMembershipModal =
+    '[data-testid="cancel-membership-modal"]';
+
+  private readonly cancelMembershipButton =
+    'button[data-testid="cancel-membership-modal-submit-button"]';
+
+  private readonly addFundsModal = '[data-testid="add-funds-modal"]';
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+  }
+
+  /**
+   * Check if the Shield Detail page is loaded
+   */
+  async checkPageIsLoaded(): Promise<void> {
+    console.log('Checking Shield Detail page is loaded with selectors:', [
+      this.pageContainer,
+      this.membershipStatus,
+    ]);
+    await this.driver.waitForMultipleSelectors([
+      this.pageContainer,
+      this.membershipStatus,
+    ]);
+    console.log('Shield Detail page is loaded');
+  }
+
+  /**
+   * Get the membership status text
+   */
+  async getMembershipStatus(): Promise<string> {
+    console.log('Getting membership status');
+    const element = await this.driver.findElement(this.membershipStatus);
+    const membershipStatusText = await element.getText();
+    console.log('Membership status text:', membershipStatusText);
+    return membershipStatusText;
+  }
+
+  /**
+   * Get the customer ID text
+   */
+  async getCustomerId(): Promise<string> {
+    console.log('Getting customer ID');
+    const element = await this.driver.findElement(this.customerId);
+    const customerIdText = await element.getText();
+    console.log('Customer ID text:', customerIdText);
+    return customerIdText;
+  }
+
+  /**
+   * Check if trial tag is displayed
+   */
+  async checkTrialTagDisplayed(): Promise<void> {
+    console.log('Checking trial tag is displayed');
+    await this.driver.waitForSelector(this.trialTag);
+  }
+
+  /**
+   * Check if paused tag is displayed
+   */
+  async checkPausedTagDisplayed(): Promise<void> {
+    console.log('Checking paused tag is displayed');
+    await this.driver.waitForSelector(this.pausedTag);
+  }
+
+  /**
+   * Click the View Full Benefits button
+   */
+  async clickViewBenefitsButton(): Promise<void> {
+    console.log('Clicking View Full Benefits button');
+    await this.driver.clickElement(this.viewBenefitsButton);
+  }
+
+  /**
+   * Click the Submit Case button
+   */
+  async clickSubmitCaseButton(): Promise<void> {
+    console.log('Clicking Submit Case button');
+    await this.driver.clickElement(this.submitCaseButton);
+  }
+
+  /**
+   * Click the Resubscribe button
+   */
+  async clickResubscribeButton(): Promise<void> {
+    console.log('Clicking Resubscribe button');
+    await this.driver.clickElement(this.resubscribeButton);
+  }
+
+  /**
+   * Click the Cancel Membership button
+   */
+  async clickCancelButton(): Promise<void> {
+    console.log('Clicking Cancel Membership button');
+    await this.driver.clickElement(this.cancelButton);
```

Comment:
I'm unfamiliar with the feature, but does this action close a modal? if so, we could use this instead `clickElementAndWaitToDisappear` to avoid flakiness